timeseries/src/autogluon/timeseries/models/local/abstract_local_model.py

```diff
@@ -0,0 +1,133 @@
+from autogluon.timeseries.utils.seasonality import get_seasonality
+
+logger = logging.getLogger(__name__)
+
+
+class AbstractLocalModel(AbstractTimeSeriesModel):
+    allowed_local_model_args: List[str] = []
+
+    def __init__(
+        self,
+        freq: Optional[str] = None,
+        prediction_length: int = 1,
+        path: Optional[str] = None,
+        name: Optional[str] = None,
+        eval_metric: str = None,
+        hyperparameters: Dict[str, Any] = None,
+        **kwargs,  # noqa
+    ):
+        name = name or re.sub(r"Model$", "", self.__class__.__name__)
+        super().__init__(
+            path=path,
+            freq=freq,
+            prediction_length=prediction_length,
+            name=name,
+            eval_metric=eval_metric,
+            hyperparameters=hyperparameters,
+            **kwargs,
+        )
+        if hyperparameters is None:
+            hyperparameters = {}
+        n_jobs = hyperparameters.get("n_jobs", 0.5)
+        if isinstance(n_jobs, float) and 0 < n_jobs <= 1.0:
+            self.n_jobs = max(int(cpu_count() * n_jobs), 1)
+        elif isinstance(n_jobs, int):
+            self.n_jobs = n_jobs
+        else:
+            raise ValueError(f"n_jobs must be a float between 0 and 1 or an integer (received n_jobs = {n_jobs})")
+        self._local_model_args: Dict[str, Any] = None
+        self._cached_predictions: Dict[str, pd.DataFrame] = {}
+
+    def _fit(self, train_data: TimeSeriesDataFrame, time_limit: int = None, **kwargs):
+        self._check_fit_params()
+        # Initialize parameters passed to each local model
+        raw_local_model_args = self._get_model_params().copy()
+        raw_local_model_args.pop("n_jobs", None)
+
+        unused_local_model_args = []
+        local_model_args = {}
+        for key, value in raw_local_model_args.items():
+            if key in self.allowed_local_model_args:
+                local_model_args[key] = value
+            else:
+                unused_local_model_args.append(key)
+
+        if len(unused_local_model_args):
+            logger.warning(
+                f" {self.name} ignores following hyperparameters: {unused_local_model_args}. "
+                f"See the docstring of {self.name} for the list of supported hyperparameters."
+            )
+
+        if "seasonal_period" not in local_model_args or local_model_args["seasonal_period"] is None:
+            local_model_args["seasonal_period"] = get_seasonality(train_data.freq)
+        local_model_args["freq"] = train_data.freq
+        self.freq = train_data.freq
+
+        self._local_model_args = self._update_local_model_args(local_model_args=local_model_args, data=train_data)
+
+        logger.debug(f"{self.name} is a local model, so the model will be fit at prediction time.")
+        return self
+
+    def _update_local_model_args(self, data: TimeSeriesDataFrame, local_model_args: Dict[str, Any]) -> Dict[str, Any]:
```

Comment:
For pretty much all other models (except SeasonalNaive and Naive), this will depend on the dataset (e.g., its frequency), so I chose to make this explicit.